src/primitives.es6

```diff
@@ -274,6 +378,16 @@ export class ArrayType extends Collection {
 
     super('array', converted, meta, attributes);
   }
+
+  contains(value) {
+    for (let item of this.content) {
+      if (item.toValue() === value) {
+        return true;
```

Comment:
I've added code for this, both deep equality and `hasKey`. 

Now, `equals` also deep equality.